go/vt/vtgate/vtgateconn/vtgateconn.go

```diff
@@ -144,7 +144,11 @@ func (sn *VTGateSession) StreamExecute(ctx context.Context, query string, bindVa
 	// StreamExecute is only used for SELECT queries that don't change
```

Comment:
Is this still true?

Reply:
```
	// StreamExecute is only used for SELECT queries that don't change
	// the session. So, the protocol doesn't return an updated session.
	// This may change in the future.
```

For sure it has changed and this is the future :)